src/common/daft-config/src/lib.rs

```diff
@@ -2,13 +2,17 @@
 pub struct DaftConfig {
     pub merge_scan_tasks_min_size_bytes: usize,
     pub merge_scan_tasks_max_size_bytes: usize,
+    pub broadcast_join_size_bytes_threshold: usize,
 }
 
 impl Default for DaftConfig {
     fn default() -> Self {
         DaftConfig {
-            merge_scan_tasks_min_size_bytes: 64 * 1024 * 1024, // 64MB
-            merge_scan_tasks_max_size_bytes: 512 * 1024 * 1024, // 512MB
+            merge_scan_tasks_min_size_bytes: 64 * 1024 * 1024, // 64 MiB
+            merge_scan_tasks_max_size_bytes: 512 * 1024 * 1024, // 512 MiB
+            broadcast_join_size_bytes_threshold: std::env::var("DAFT_BROADCAST_JOIN_THRESHOLD")
```

Comment:
Should we remove this env var?

Could be good to standardize in the future on something like `DAFT_CFG__*` for setting configs via the environment, but until then I think having only one way to do it `daft.context.set_config()` is maybe a better idea so we can avoid having too many (undocumented!) env variable flags.

Reply:
Sure, I mostly added this to be able to tweak the threshold without code modifications during benchmarking, since environment variables are a good bit easier to inject, and this environment variable being undocumented + having the `DAFT_` prefix essentially makes it a private/developer convenience API.

We can ultimately do something similar to what we did for Ray, e.g. a
```rust
from_env_with_default!(broadcast_join_threshold, usize, 10 * 1024 * 1024)
```
macro that uses the (parsed) `DAFT_BROADCAST_JOIN_THRESHOLD` environment variable if its set, and otherwise falls back to the provided default.

Reply:
https://github.com/ray-project/ray/blob/master/src/ray/common/ray_config_def.h
https://github.com/ray-project/ray/blob/e862ec27d8eb7bee9bd151915de989a49b3c3fc0/src/ray/common/ray_config.h#L72-L90